### ISSUE TITLE
Are the anchor values correct in the config file?

### ISSUE THREAD

**Issue by Itto1992** [argusswift/YOLOv4-pytorch#157]
Thanks for your creating this great repo! It really helps me understand the implementation details of YOLOv4.

Let me ask about the anchor values defined in `config/yolov4_config.py`.

https://github.com/argusswift/YOLOv4-pytorch/blob/158ea9223cb57f88d81f60dd894b7be9f1a285f8/config/yolov4_config.py#L168-L185

In my understanding, width (and height) of anchors is calculated by `stride * width_ratio (height_ratio)`. Thus, I think the anchor sizes for small objects should be `[(10, 13), (16, 30), (33, 23)]`. However, the comment says `# Anchors for small obj(12,16),(19,36),(40,28)`. (The anchors for middle and large objects are also the same.)
Could you tell me how the values are calculated?

Thanks!

**Comment by jingtianyilong** [comment]
They are different for different scales.
For small you devide by 8 then middle you divide them by 16 and for large you divide them by 32.
see https://github.com/jingtianyilong/YOLOv4-pytorch/blob/master/get_anchor.py

**Comment by Itto1992** [comment]
Thanks for your rapid reply!

Sorry for my confusing writing. What I want to point out is the mismatch between `MODEL['ANCHORS']` values and the commented anchor values. In other words, `MODEL['ANCHORS'][0] * 8` does not recover `(12,16),(19,36),(40,28)` for example. I suspect one of them is incorrect.

**Comment by jingtianyilong** [comment]
> Thanks for your rapid reply!
> 
> Sorry for my confusing writing. What I want to point out is the mismatch between `MODEL['ANCHORS']` values and the commented anchor values. In other words, `MODEL['ANCHORS'][0] * 8` does not recover `(12,16),(19,36),(40,28)` for example. I suspect one of them is incorrect.

It won't matter that much. Anchor is only a starting point. So it won't affect the training that much as long as they are not too extreme. I don't know where did these number comes from since I don't even use COCO. But in general, I would suggest that you calculate the anchor by yourself.

**Comment by Itto1992** [comment]
> Yeah, but it still looks fine to me. Anchor is only a starting point. So it won't affect the training that much as long as they are not too extreme.

I agree with you. I feel refreshed since my understanding is not wrong.

Thank you!
